src/multi.c

```diff
@@ -382,21 +382,17 @@ int isWatchedKeyExpired(client *c) {
 /* "Touch" a key, so that if this key is being WATCHed by some client the
  * next EXEC will fail. */
 void touchWatchedKey(redisDb *db, robj *key) {
-    list *clients;
-    listIter li;
-    listNode *ln;
+    list *clients = NULL;
 
     if (dictSize(db->watched_keys) == 0) return;
-    clients = dictFetchValue(db->watched_keys, key);
-    if (!clients) return;
 
     /* Mark all the clients watching this key as CLIENT_DIRTY_CAS */
     /* Check if we are already watching for this key */
-    listRewind(clients,&li);
-    while((ln = listNext(&li))) {
-        client *c = listNodeValue(ln);
-
+    while ((clients = dictFetchValue(db->watched_keys, key)) && listLength(clients) != 0) {
```

Comment:
I don't believe listLength() can be zero, so let's either assert it or drop the conditional.

Reply:
Right, Removed.